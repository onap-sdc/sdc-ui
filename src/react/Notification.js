import React from 'react';
import PropTypes from 'prop-types';
import SVGIcon from './SVGIcon';

const Notification = ({
    title,
    message,
    type,
    onClick,
    className,
    dataTestId
}) => {
    return (
        <div
            className={`sdc-notification type-${type} ${className}`}
            data-test-id={dataTestId}
            onClick={onClick}>
            <div className="sdc-notification__icon_container">
                <SVGIcon name="check" className="sdc-notification_svg-icon" />
            </div>
            <div className="sdc-notification__message">
                <div className="sdc-notification__title">{title}</div>
                <div className="sdc-notification__text">{message}</div>
            </div>
        </div>
    );
};

Notification.propTypes = {
    title: PropTypes.string,
    message: PropTypes.string,
    type: PropTypes.string,
    dataTestId: PropTypes.string
};

Notification.defaultProps = {
    className: '',
    type: 'info',
    title: '',
    message: '',
    dataTestId: ''
};

export default Notification;
