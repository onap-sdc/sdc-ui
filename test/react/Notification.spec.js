import React from 'react';
import Notification from '../../src/react/Notification.js';

import renderer from 'react-test-renderer';

describe('Notification', () => {
	test('Notification - info', () => {
		const button = renderer.create(<Notification type='info' message='message' title='title' />).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Notification - success', () => {
		const button = renderer.create(<Notification type='success' message='message' title='title' />).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Notification - warning', () => {
		const button = renderer.create(<Notification type='warning' message='message' title='title' />).toJSON();
		expect(button).toMatchSnapshot();
	});

	test('Notification - error', () => {
		const button = renderer.create(<Notification type='error' message='message' title='title' />).toJSON();
		expect(button).toMatchSnapshot();
	});
});