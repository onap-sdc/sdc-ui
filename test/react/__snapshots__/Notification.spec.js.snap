// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Notification - error 1`] = `
<div
  className="sdc-notification "
  data-test-id=""
  onClick={undefined}
>
  <div
    className="sdc-notification__wrapper type-error "
  >
    <div
      className="sdc-notification__content"
    >
      <div
        className="sdc-nitification__icon_container"
      >
        <div
          className="svg-icon-wrapper  sdc-notification_svg-icon   bottom"
          disabled={undefined}
          onClick={undefined}
        >
          <svg />
          
        </div>
      </div>
      <div
        className="sdc-notification__message"
      >
        <div
          className="sdc-notification__title"
        >
          title
        </div>
        <div
          className="sdc-notification__text"
        >
          message
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Notification Notification - info 1`] = `
<div
  className="sdc-notification "
  data-test-id=""
  onClick={undefined}
>
  <div
    className="sdc-notification__wrapper type-info "
  >
    <div
      className="sdc-notification__content"
    >
      <div
        className="sdc-nitification__icon_container"
      >
        <div
          className="svg-icon-wrapper  sdc-notification_svg-icon   bottom"
          disabled={undefined}
          onClick={undefined}
        >
          <svg />
          
        </div>
      </div>
      <div
        className="sdc-notification__message"
      >
        <div
          className="sdc-notification__title"
        >
          title
        </div>
        <div
          className="sdc-notification__text"
        >
          message
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Notification Notification - success 1`] = `
<div
  className="sdc-notification "
  data-test-id=""
  onClick={undefined}
>
  <div
    className="sdc-notification__wrapper type-success "
  >
    <div
      className="sdc-notification__content"
    >
      <div
        className="sdc-nitification__icon_container"
      >
        <div
          className="svg-icon-wrapper  sdc-notification_svg-icon   bottom"
          disabled={undefined}
          onClick={undefined}
        >
          <svg />
          
        </div>
      </div>
      <div
        className="sdc-notification__message"
      >
        <div
          className="sdc-notification__title"
        >
          title
        </div>
        <div
          className="sdc-notification__text"
        >
          message
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Notification Notification - warning 1`] = `
<div
  className="sdc-notification "
  data-test-id=""
  onClick={undefined}
>
  <div
    className="sdc-notification__wrapper type-warning "
  >
    <div
      className="sdc-notification__content"
    >
      <div
        className="sdc-nitification__icon_container"
      >
        <div
          className="svg-icon-wrapper  sdc-notification_svg-icon   bottom"
          disabled={undefined}
          onClick={undefined}
        >
          <svg />
          
        </div>
      </div>
      <div
        className="sdc-notification__message"
      >
        <div
          className="sdc-notification__title"
        >
          title
        </div>
        <div
          className="sdc-notification__text"
        >
          message
        </div>
      </div>
    </div>
  </div>
</div>
`;
